mobile/src/main/java/ch/epfl/sdp/mobile/ui/social/FollowButton.kt

```diff
@@ -0,0 +1,87 @@
+package ch.epfl.sdp.mobile.ui.social
+
+import androidx.compose.animation.*
+import androidx.compose.animation.core.updateTransition
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.mobile.state.LocalLocalizedStrings
+import ch.epfl.sdp.mobile.ui.FadeEnterTransition
+import ch.epfl.sdp.mobile.ui.FadeExitTransition
+import ch.epfl.sdp.mobile.ui.PawniesColors
+
+/**
+ * An implementation of [ButtonColors] used to change the color of a button according to whether it
+ * is selected.
+ *
+ * @param selected a [State] of [Boolean] indicating whether the button is selected.
+ */
+private class FollowButtonColors(selected: State<Boolean>) : ButtonColors {
+
+  /** A [State] of [Boolean] indicating whether the button is selected. */
+  private val selected by selected
```

Comment:
Question : How can you assign a variable with the name ?

Reply:
Not sure I completely understood but the other selected name refers to the class parameter

Reply:
Indeed, this is a property delegate for the class parameter `selected: State<Boolean>`.